### PR TITLE
istioctl: add default name for waypoint and remove leftover service-account code

### PR DESCRIPTION
**Pull request by MorrisLaw**
**Please provide a description of this PR:**
Adds a default name for waypoint and removes leftover service-account code that was missed [here](https://github.com/istio/istio/pull/50221).

Resolves: #49915

### CODE REVIEW

**Review by bleggett — COMMENTED** [review]
Given we will conflict with both `w` and `n`,  and we don't have any other 2-char aliases, so adding `wn` is a little weird - should we just make supplying the name mandatory, rather than an optional `-n/-wn` flag? 

The "default to svcacctname" thing isn't all that useful or intuitive now with the new/upcoming capture model anyway, so I don't think we need to keep that.

We are also going to change waypoint creation to create unique per-waypoint service accounts by default anyway with that, IIRC.

We can just require people to declare a name for their waypoint when creating one, and use it.

cc @ilrudie for thoughts

**Comment by ilrudie** [comment]
yeah, defaulting to service account (or just naming it `namespace` if there's no sa) seems less great now. I think requiring a name does make some sense to me.

**Comment by MorrisLaw** [comment]
> Given we will conflict with both `w` and `n`, and we don't have any other 2-char aliases, so adding `wn` is a little weird - should we just make supplying the name mandatory, rather than an optional `-n/-wn` flag?
> 
> The "default to svcacctname" thing isn't all that useful or intuitive now with the new/upcoming capture model anyway, so I don't think we need to keep that.
> 
> We are also going to change waypoint creation to create unique per-waypoint service accounts by default anyway, IIRC.
> 
> We can just require people to declare a name for their waypoint when creating one, and use it for the svcaccount we will create.
> 
> cc @ilrudie for thoughts

TIL, the reason you don't see it is because it's not allowed: `"wn" shorthand is more than one ASCII character` is the error in those failed tests that first came up for this PR

File: istioctl/pkg/waypoint/waypoint.go [hunk 1]
```diff
@@ -56,26 +56,22 @@ const (
...
-		return name
-	}
-	makeGateway := func(forApply bool) *gateway.Gateway {
+	var waypointServiceAccount, waypointName string
+	makeGateway := func(forApply bool) (*gateway.Gateway, error) {
 		ns := ctx.NamespaceOrDefault(ctx.Namespace())
 		if ctx.Namespace() == "" && !forApply {
 			ns = ""
 		}
+		if waypointName == "" {
```
Comment (howardjohn) [inline istioctl/pkg/waypoint/waypoint.go:65]: I think a default would probably be nice? Especially if we assume a typical user will have one per namespace, having a convention for the name of that could be handy

Comment (hanxiaop) [reply istioctl/pkg/waypoint/waypoint.go:65]: +1 for a default value.

**Review by hanxiaop — COMMENTED** [review]
Can you also add a test covering the name flag? There are integration tests around the waypoint command.

File: istioctl/pkg/waypoint/waypoint.go (hunk 1 above)
Comment (bleggett) [reply istioctl/pkg/waypoint/waypoint.go:65]: rather than throw an error we can just do "no name specified, using `{namespace}-waypoint`" yea

**Comment by bleggett** [comment]
> > Given we will conflict with both `w` and `n`, and we don't have any other 2-char aliases, so adding `wn` is a little weird - should we just make supplying the name mandatory, rather than an optional `-n/-wn` flag?
> > The "default to svcacctname" thing isn't all that useful or intuitive now with the new/upcoming capture model anyway, so I don't think we need to keep that.
> > We are also going to change waypoint creation to create unique per-waypoint service accounts by default anyway, IIRC.
> > We can just require people to declare a name for their waypoint when creating one, and use it for the svcaccount we will create.
> > cc @ilrudie for thoughts
> 
> TIL, the reason you don't see it is because it's not allowed: `"wn" shorthand is more than one ASCII character` is the error in those failed tests that first came up for this PR

Right, which makes sense - 2 char short args are confusing, and in this case the difference between 2 and 4 chars for the full flag is arguably trivial, so single-char args are a convention it's reasonable to keep, especially in this case where I don't think we need an optional arg.

File: istioctl/pkg/waypoint/waypoint.go (hunk 1 above)
Comment (howardjohn) [reply istioctl/pkg/waypoint/waypoint.go:65]: Note the class name is appended to the generated things. So if we keep `istio-waypoint` class, `name-waypoint-istio-waypoint` would be the result which is pretty gross.

**Review by howardjohn — APPROVED** [review]
LGTM, lets followup with a default though

**Comment by howardjohn** [comment]
@MorrisLaw fyi needs a rebase

**Pull request merged**